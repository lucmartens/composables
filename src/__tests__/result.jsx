import React from 'react';
import { mount } from 'enzyme';

import Result from '../result';

describe('Result converts promise to', () => {
  const tick = () => {
    return new Promise(resolve => setTimeout(resolve, 0));
  };

  test('pending result object', () => {
    const render = jest.fn();
    mount(<Result promise={new Promise(() => {})} render={render} />);

    expect(render.mock.calls.length).toBe(1);
    expect(render.mock.calls[0][0]).toMatchSnapshot();
  });

  test('successful result object', async () => {
    const render = jest.fn();
    mount(<Result promise={Promise.resolve('value')} render={render} />);

    expect(render.mock.calls.length).toBe(1);
    await tick();
    expect(render.mock.calls.length).toBe(2);
    expect(render.mock.calls[1][0]).toMatchSnapshot();
  });

  test('failed result object', async () => {
    const render = jest.fn();
    mount(<Result promise={Promise.reject('error_code')} render={render} />);

    expect(render.mock.calls.length).toBe(1);
    await tick();

    expect(render.mock.calls.length).toBe(2);
    expect(render.mock.calls[1][0]).toMatchSnapshot();
  });
});
