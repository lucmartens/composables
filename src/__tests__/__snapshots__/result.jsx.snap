// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result converts promise to failed result object 1`] = `
Object {
  "error": true,
  "value": "error_code",
}
`;

exports[`Result converts promise to pending result object 1`] = `
Object {
  "pending": true,
  "value": undefined,
}
`;

exports[`Result converts promise to successful result object 1`] = `
Object {
  "done": true,
  "value": "value",
}
`;
